lardataobj/RawData/raw.cxx

```diff
@@ -1173,4 +1200,207 @@ namespace raw {
+        if (standard_fibonacci) {
+          if      (d==1 ) {add_to_sequence_terminate({1            }, sqce, cmp); break;}
+          else if (d==2 ) {add_to_sequence_terminate({0,1          }, sqce, cmp); break;}
+          else if (d==3 ) {add_to_sequence_terminate({0,0,1        }, sqce, cmp); break;}
+          else if (d==4 ) {add_to_sequence_terminate({1,0,1        }, sqce, cmp); break;}
+          else if (d==5 ) {add_to_sequence_terminate({0,0,0,1      }, sqce, cmp); break;}
+          else if (d==6 ) {add_to_sequence_terminate({1,0,0,1      }, sqce, cmp); break;}
+          else if (d==7 ) {add_to_sequence_terminate({0,1,0,1      }, sqce, cmp); break;}
+          else if (d==8 ) {add_to_sequence_terminate({0,0,0,0,1    }, sqce, cmp); break;}
+          else if (d==9 ) {add_to_sequence_terminate({1,0,0,0,1    }, sqce, cmp); break;}
+          else if (d==10) {add_to_sequence_terminate({0,1,0,0,1    }, sqce, cmp); break;}
+          else if (d==11) {add_to_sequence_terminate({0,0,1,0,1    }, sqce, cmp); break;}
+          else if (d==12) {add_to_sequence_terminate({1,0,1,0,1    }, sqce, cmp); break;}
+          else if (d==13) {add_to_sequence_terminate({0,0,0,0,0,1  }, sqce, cmp); break;}
+          else if (d==14) {add_to_sequence_terminate({1,0,0,0,0,1  }, sqce, cmp); break;}
+          else if (d==15) {add_to_sequence_terminate({0,1,0,0,0,1  }, sqce, cmp); break;}
+          else if (d==16) {add_to_sequence_terminate({0,0,1,0,0,1  }, sqce, cmp); break;}
+          else if (d==17) {add_to_sequence_terminate({1,0,1,0,0,1  }, sqce, cmp); break;}
+          else if (d==18) {add_to_sequence_terminate({0,0,0,1,0,1  }, sqce, cmp); break;}
+          else if (d==19) {add_to_sequence_terminate({1,0,0,1,0,1  }, sqce, cmp); break;}
+          else if (d==20) {add_to_sequence_terminate({0,1,0,1,0,1  }, sqce, cmp); break;}
+          else if (d==21) {add_to_sequence_terminate({0,0,0,0,0,0,1}, sqce, cmp); break;}
+          else if (d==22) {add_to_sequence_terminate({1,0,0,0,0,0,1}, sqce, cmp); break;}
+          else if (d==23) {add_to_sequence_terminate({0,1,0,0,0,0,1}, sqce, cmp); break;}
+          else if (d==24) {add_to_sequence_terminate({0,0,1,0,0,0,1}, sqce, cmp); break;}
+          else if (d==25) {add_to_sequence_terminate({1,0,1,0,0,0,1}, sqce, cmp); break;}
+          else if (d==26) {add_to_sequence_terminate({0,0,0,1,0,0,1}, sqce, cmp); break;}
+          else if (d==27) {add_to_sequence_terminate({1,0,0,1,0,0,1}, sqce, cmp); break;}
+          else if (d==28) {add_to_sequence_terminate({0,1,0,1,0,0,1}, sqce, cmp); break;}
+          else if (d==29) {add_to_sequence_terminate({0,0,0,0,1,0,1}, sqce, cmp); break;}
+          else if (d==30) {add_to_sequence_terminate({1,0,0,0,1,0,1}, sqce, cmp); break;}
+          else if (d==31) {add_to_sequence_terminate({0,1,0,0,1,0,1}, sqce, cmp); break;}
+        }
+      
+        // Find the first Fibonacci number that exceed the number
+        for (; iFib<FibNumbers.size(); ++iFib) {
```

Comment:
consider using std::lower_bound here as it does a binary search with logarithmic time instead of linear time